src/services/Audius/helpers.ts

```diff
@@ -178,3 +179,30 @@ export async function getContentNodeMetadata(
     return await getNodeMetadata(endpoint)
   }
 }
+
+export function decodeCallData(types: string[], callData: string) {
+  // TODO: Like methods above and throughout, move to better pattern
+  const web3 = window.audiusLibs.ethWeb3Manager.web3
+  return web3.eth.abi.decodeParameters(types, callData)
+}
+
+export function decodeProposalCallData(proposal: Proposal) {
+  const signatureSplit = proposal.functionSignature.split('(')
+  const functionName = signatureSplit?.[0]
+
+  const types = signatureSplit?.[1]?.split(')')?.[0]?.split(',')
+  const decoded: { [key: string]: string } = AudiusClient.decodeCallData(
+    types,
```

Comment:
`types` might be undefined right? might be worth adding a check for that but idk how likely that is

Reply:
ty